### PR TITLE
Voice mask

### PR DESCRIPTION
**Pull request by vulppine**
## About the PR 

Voice masking. Allows a user to change their voice at will. Also adds the voice mask - a stealth tool for antagonists that changes their voice to whatever they set using the voice mask UI.

**Changelog**

:cl:
- add: Voice masking is now possible for mobs that speak.
- add: The Syndicate has found a supplier for voice modification tools, and is now supplying them to their agents. Be aware of who you're speaking to: they might not be who they sound like.

### CODE REVIEW

File: Content.Server/VoiceMask/VoiceMaskerSystem.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+using Content.Shared.Inventory.Events;
+
+namespace Content.Server.VoiceMask;
+
+public sealed class VoiceMaskerSystem : EntitySystem
+{
+    public override void Initialize()
```
Comment (metalgearsloth) [inline Content.Server/VoiceMask/VoiceMaskerSystem.cs:7]: Can probably just make this a partial class of VoiceMaskSystem.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by vulppine** [comment]
> Would you like to give toggleable feature on syndicate gas mask to merely scramble your voice to Unknown (or something like that), complete with spooky robotic speech sound? That would not hide that you are baddie, but saves you from having to mime out your demands.

right now, just taking off your ID will set the name you speak with to the generic name you have in identity

**Comment by vulppine** [comment]
actually, having the identity name is probably too revealing sooooooo

**Comment by vulppine** [comment]
there was some discussion on discord about the state of this feature, so i'll summarize it here:

voice masks (or, voice changers) currently mimic the SS13 functionality of voice changers, but it's not inherently intuitive for newer players, as it requires usage of the ID card currently equipped by the player - so, a proposed change was made that instead allows the player to set their current voice via a dialog

i don't necessarily agree with this, but i also implemented the functionality so that if some consensus is reached, it shouldn't take very long to use the new set of changes

pro and con list, feature wise:

**SS13-style: ID cards as the voice mask name dependency**
- Pros:
  - Cheaper (1 - 2 TC balance because of how difficult it is to change identities)
  - Requires more interaction by the player in order to get an ID card if they forgo the agent card
  - Requires more book-keeping by the player in order to keep a distinct set of identities
- Cons:
  - Not intuitive (requires knowledge that ID card will change the name, but instructions on this are filled in on the uplink description
  - Not realistic (doesn't make sense that the ID card dictates the voice generated by some other non-related item)
  - Requires more book-keeping by the player in order to keep a set of identities (managing ID cards is also annoying)

**Proposed change: Voice mask name is stored in the player/mask**
- Pros:
  - More intuitive - player has gas mask, adjusts name, and uses it
  - More realistic - voice mask generates the player's voice without being dependent on other items
  - No bookkeeping - requires no ID card, so no bookkeeping of ID cards by the player is required
  - Flexible - no ID card requirement means that the player can speak like somebody else while looking (close enough) to somebody else
- Cons:
  - Expensive (3 - 5 TC balance due to its ease of use)
  - Requires less (or zero) interaction by the player to mask a voice (player can just buy gas mask without having to deal with obtaining an ID card to use)
  - Flexible - no ID requirement means that voice names can be changed on a fly, making it less difficult for a player to keep up with identity changes

keep in mind that in SS13, most servers bundle the voice changer as part of a full "chameleon kit", which costs about 2 TC, and includes a full set of chameleon gear - the proposed change above would instead keep it as a separate item from the kit (unless it otherwise is bundled in by another PR)

**Comment by vulppine** [comment]
> With my proposal about suspicious toolbox mask getting _scrambler_, scope would be cheap identity hiding that does _not_ hide I am bad person when I am in disguise, but saves my primary identity until I slip up somehow all while allowing me to ransom station and such.

this seems like a decent tradeoff, especially since if the role/person you're impersonating usually has something that hides their face to pair with the voice changer (when trying to be stealthy)

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**